src/main/java/roomescape/interceptor/RoleCheckInterceptor.java

```diff
@@ -0,0 +1,43 @@
+package roomescape.interceptor;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.web.servlet.HandlerInterceptor;
+import roomescape.domain.member.domain.Member;
+import roomescape.domain.member.error.exception.MemberErrorCode;
+import roomescape.domain.member.error.exception.MemberException;
+import roomescape.domain.member.service.MemberService;
+
+import java.util.Arrays;
+
+public class RoleCheckInterceptor implements HandlerInterceptor {
+
+    private static final String TOKEN = "token";
+
+    private final MemberService memberService;
+
+    public RoleCheckInterceptor(MemberService memberService) {
+        this.memberService = memberService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        Cookie[] cookies = request.getCookies();
+        String token = extractTokenFromCookie(cookies);
+        Member member = memberService.findByToken(token);
+        if (member == null || !member.getRole().equals("ADMIN")) {
```

Comment:
```suggestion
        if (member == null || !member.isAdmin()) {
```

아래 글 그리고 글에서 나온 키워드를 참고해 보세요.

- [디미터 법칙(Law of Demeter)과 묻지 말고 시켜라(Tell, Don’t Ask)](https://prohannah.tistory.com/204)